modules/setting/config_env.go

```diff
@@ -177,7 +177,10 @@ func InitGiteaEnvVars() {
 	// But git would try "XDG_CONFIG_HOME/git/config" first if "HOME/.gitconfig" does not exist,
 	// then our git.InitFull would still write to "XDG_CONFIG_HOME/git/config" if XDG_CONFIG_HOME is set.
 	_ = os.Unsetenv("XDG_CONFIG_HOME")
+}
 
+func InitGiteaEnvVarsForTesting() {
```

Comment:
To reviewers: this change is to keep the logic the same as 1.23, it seems that we do not need to unset these "GIT_*" in production, the related code should be able to handle it.